packages/asset-swapper/src/utils/market_operation_utils/pools_cache/balancer_v2_utils_new.ts

```diff
@@ -0,0 +1,212 @@
+import { ChainId } from '@0x/contract-addresses';
+
+import { DEFAULT_WARNING_LOGGER } from '../../../constants';
+import { LogFunction } from '../../../types';
+import {
+    BALANCER_MAX_POOLS_FETCHED,
+    BALANCER_TOP_POOLS_FETCHED,
+    BALANCER_V2_SUBGRAPH_URL_BY_CHAIN,
+} from '../constants';
+import { BalancerSwapInfo, BalancerSwaps } from '../types';
+
+import { CacheValue, SwapInfoCache } from './pair_swaps_cache';
+import { SubgraphPoolDataService } from './sgPoolDataService';
+import {
+    BalancerSDK,
+    BalancerSdkConfig,
+    Network,
+    SwapTypes,
+    RouteProposer,
+    NewPath,
+    parseToPoolsDict,
+    PoolDictionary,
+    formatSequence,
+    getTokenAddressesForSwap,
+} from '@balancer-labs/sdk';
+
+// tslint:disable-next-line:custom-no-magic-numbers
+const ONE_DAY_MS = 24 * 60 * 60 * 1000;
+
+export interface BalancerPoolResponse {
+    poolType: string;
+    id: string;
+    tokens: Array<{ address: string; }>;
+    tokensList: string[];
+}
+
+export class BalancerV2SwapInfoCache extends SwapInfoCache {
+    routeProposer: RouteProposer;
+    poolDataService: SubgraphPoolDataService;
+
+    constructor(
+        chainId: ChainId,
+        private readonly subgraphUrl: string = BALANCER_V2_SUBGRAPH_URL_BY_CHAIN[chainId],
+        private readonly maxPoolsFetched: number = BALANCER_MAX_POOLS_FETCHED,
+        private readonly _topPoolsFetched: number = BALANCER_TOP_POOLS_FETCHED,
+        private readonly _warningLogger: LogFunction = DEFAULT_WARNING_LOGGER,
+        cache: { [key: string]: CacheValue } = {},
+    ) {
+        super(cache);
+        const config: BalancerSdkConfig = {
+            network: Network[ChainId[chainId]],
+            rpcUrl: `https://mainnet.infura.io/v3/${process.env.INFURA}`,
```

Comment:
Is there a way to pass in a generic provider object? I.e., something that just implements an interface like `{ send(payload); }`

Reply:
Unfortunately there's not at this time but in this case the provider isn't actually used within the SDK so you could just do: `rpcUrl: '',`.